### PR TITLE
tests: adding test to ensure no migration can collide with 4.x upgrad…

### PR DESCRIPTION
**Pull request by goutnet**
## Description

Following conversation on slack, Adding a test here to ensure no new migration can be added to the application (as this would make problems on 4.x migration path).

@marksweb  @Aiky30  @vinitkumar

### CODE REVIEW

**Review by vinitkumar — APPROVED** [review]
Looks good to me. Let's get this merged and released 🚢  🚀

**Review by vinitkumar — CHANGES_REQUESTED** [review]
Please check the tests, they are failing.

File: cms/tests/test_compatibility_with_4x_branch.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import cms
```
Comment (marksweb) [inline cms/tests/test_compatibility_with_4x_branch.py:1]: This isn't being used according to the checks so can be removed 👌

**Review by marksweb — CHANGES_REQUESTED** [review]

File: cms/tests/test_compatibility_with_4x_branch.py (hunk 1 above)
Comment (goutnet) [reply cms/tests/test_compatibility_with_4x_branch.py]: thanks! (and done) :)

**Comment by goutnet** [comment]
tests still failing, let me check

**Comment by Aiky30** [comment]
@marksweb @vinitkumar Seems the settings for the CI are configured to enforce the two tests: database-mysql, database-postgres. Are you doing something behind the scenes with this as it's not in the codebase yet or else it would have ran here.

**Comment by goutnet** [comment]
Actually they just run all the tests in the different env. This is just one
test run in all scenarios :)

**Comment by goutnet** [comment]
(closed&re-open to force travis to run again)

**Comment by goutnet** [comment]
must be an issue in the config on github, let me check

**Comment by vinitkumar** [comment]
@goutnet still comes up
Like I said earlier, do this from a fresh branch, some weird github actions caching issue is going on here. Also, the fastest way to get this resolved. It's just a diff from a single file, so shouldn't take much time.

**Comment by goutnet** [comment]
not really sure what's going on on 3.6 there …

**Review by vinitkumar — APPROVED** [review]

**Pull request merged**